### PR TITLE
Config option to select which job framework integrations should be enabled

### LINKED ISSUE #665: Error messages when MPIJob CRD is not installed
**What happened**:

Even if a user doesn't plan to use MPIJob, they might observe errors in the logs, which can be confusing.

```
{"level":"error","ts":"2023-03-29T17:21:00.928174568Z","logger":"controller-runtime.source","caller":"source/source.go:143","msg":"if kind is a CRD, it should be installed before calling Start","kind":"[MPIJob.kubeflow.org](http://mpijob.kubeflow.org/)","error":"no matches for kind \"MPIJob\" in version \"[kubeflow.org/v2beta1\](http://kubeflow.org/v2beta1%5C)"","stacktrace":"[sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1.1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/source/source.go:143\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:235\nk8s.io/apimachinery/pkg/util/wait.WaitForWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:662\nk8s.io/apimachinery/pkg/util/wait.poll\n\t/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:596\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:547\nsigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/source/source.go:136](http://sigs.k8s.io/controller-runtime/pkg/source.%28*Kind%29.Start.func1.1%5Cn%5Ct/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/source/source.go:143%5Cnk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext%5Cn%5Ct/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:235%5Cnk8s.io/apimachinery/pkg/util/wait.WaitForWithContext%5Cn%5Ct/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:662%5Cnk8s.io/apimachinery/pkg/util/wait.poll%5Cn%5Ct/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:596%5Cnk8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext%5Cn%5Ct/go/pkg/mod/k8s.io/apimachinery@v0.26.3/pkg/util/wait/wait.go:547%5Cnsigs.k8s.io/controller-runtime/pkg/source.%28*Kind%29.Start.func1%5Cn%5Ct/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/source/source.go:136)"}
```

**What you expected to happen**:

No errors. Perhaps this could be avoided by explicitly enabling MPI support via command line flag.

I wouldn't do this through the configuration API, as we plan to eventually move integrations into their own binaries.

**How to reproduce it (as minimally and precisely as possible)**:

1. Do not install mpi-operator
2. Install kueue.
3. See logs

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mwielgus**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Currently all framework integrations are enabled, causing error logs on startup if the corresponding CRDs are not there. This PR adds a config option to select which integrations should be enabled and defaults it to k8s job only.

#### Which issue(s) this PR fixes:

Fixes #665

### CODE REVIEW

**Comment by mwielgus** [comment]
/assign @alculquicondor 
cc @mimowo

**Review by alculquicondor — COMMENTED** [review]
As discussed offline, we can preserve the webhook configurations and short-circuit if the particular framework is disabled.

File: apis/config/v1beta1/configuration_types.go [hunk 1]
```diff
@@ -96,3 +100,8 @@ type ClientConnection struct {
 	// Burst allows extra queries to accumulate when a client is exceeding its rate.
 	Burst *int32 `json:"burst,omitempty"`
 }
+
+type Integrations struct {
+	// List of framework names to be enabled.
+	Frameworks []string `json:"frameworks,omitempty"`
```
[the remark this answers is not in the document]
Comment (mwielgus) [reply apis/config/v1beta1/configuration_types.go:106]: Done.

File: config/components/manager/controller_manager_config.yaml [hunk 2]
```diff
@@ -27,3 +27,7 @@ clientConnection:
 #  enable: false
 #  webhookServiceName: ""
 #  webhookSecretName: ""
+integrations:
+  frameworks:
+  - job
```
[the remark this answers is not in the document]
Comment (mwielgus) [reply config/components/manager/controller_manager_config.yaml:32]: Done

File: config/components/webhook/manifests.yaml
Comment (alculquicondor) [inline config/components/webhook/manifests.yaml:1]: Could you restore the original file?

File: main.go [hunk 3]
```diff
@@ -169,35 +174,52 @@ func setupControllers(mgr ctrl.Manager, cCache *cache.Cache, queues *queue.Manag
...
+			setupLog.Error(err, "unable to create controller", "controller", "Job")
+			os.Exit(1)
+		}
+		if err := job.SetupWebhook(mgr, jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName)); err != nil {
+			setupLog.Error(err, "Unable to create webhook", "webhook", "Job")
+			os.Exit(1)
+		}
+	} else {
+		if err := dummy.SetupWebhook(mgr, job.WebhookType()); err != nil {
+			setupLog.Error(err, "Unable to create webhook", "webhook", "MPIJob")
```
Comment (alculquicondor) [inline main.go:199]:
```suggestion
			setupLog.Error(err, "Unable to create webhook", "webhook", "Job")
```

File: pkg/controller/jobs/dummy/dummy_webhook.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+package dummy
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type dummyWebhook struct {
```
Comment (alculquicondor) [inline pkg/controller/jobs/dummy/dummy_webhook.go:26]:
```suggestion
type webhook struct {
```

File: main.go [hunk 5]
```diff
@@ -47,6 +47,7 @@ import (
 	"sigs.k8s.io/kueue/pkg/controller/core"
 	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
 	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/dummy"
```
[the remark this answers is not in the document]
Comment (mwielgus) [reply main.go:50]: Done.

File: main.go (hunk 3 above)
Comment (mwielgus) [reply main.go:199]: Done

File: pkg/controller/jobs/dummy/dummy_webhook.go (hunk 4 above)
Comment (mwielgus) [reply pkg/controller/jobs/dummy/dummy_webhook.go:26]: Done.

**Comment by alculquicondor** [comment]
> /retest

Please make sure there is an open issue for a flaky test before retesting

**Pull request merged**